docs/book/src/quinn/data-transfer.md

```diff
@@ -0,0 +1,167 @@
+async fn iterate_streams(mut connection: NewConnection) -> anyhow::Result<()> {
+    // Iterate unidirectional streams with only the receiving side.
+    while let Some(Ok(recv)) = connection.uni_streams.next().await { }
+    // Iterate bidirectional streams with both sent and receiving side.
+    while let Some(Ok((sent, recv))) = connection.bi_streams.next().await { }
+    // Iterate arrived datagrams.
+    while let Some(Ok(bytes)) = connection.datagrams.next().await { }
+
+    Ok(())
+}
+```
+* *(Note that this example would get stuck in the first while loop)* 
+
+*Open various kinds of streams*
+
+```rust
+async fn open_streams(mut connection: Connection) -> anyhow::Result<()> {
+    // Open unidirectional stream.
+    let mut send = connection.
+        open_uni()
+        .await?;
+
+    // Open bidirectional stream.
+    let (send, recv) = connection.
+        open_bi()
+        .await?;
+
+    Ok(())
+}
+```
+
+## Bidirectional Streams
+
+With bidirectional streams data can be sent in both directions, for example, from the connection initiator to the peer and the other way around.
+ 
+*open bidirectional stream*
+
+```rust
+async fn open_bidirectional_stream(connection: Connection) -> anyhow::Result<()> {
+    let (mut send, recv) = connection.
+        open_bi()
+        .await?;
+
+    send.write_all(b"test").await?;
+    send.finish().await?;
+    
+    let received = recv.read_to_end(10).await?;
+
+    Ok(())
+}
+```
+
+*iterate incoming bidirectional stream(s)*
+
+```rust
+async fn receive_bidirectional_stream(mut connection: NewConnection) -> anyhow::Result<()> {
+    while let Some(Ok((sent, recv))) = connection.bi_streams.next().await {
+        // Because it is a bidirectional stream, we can both sent and recieve.
+        println!("request: {:?}", recv.read_to_end(50).await?);
+
+        send.write_all(b"response").await?;
+        send.finish().await?;
+    }
+
+    Ok(())
+}
+```
+
+## Unidirectional Streams 
+
+With unidirectional streams, you can carry data only in one direction: from the initiator of the stream to its peer.
+    
+*open unidirectional stream*
+
+```rust
+async fn open_unidirectional_stream(connection: Connection)-> anyhow::Result<()> {
+    let mut send = connection.
+        open_uni()
+        .await?;
+
+    send.write_all(b"test").await.unwrap();
+    send.finish().await?;
+
+    Ok(())
+}
+```
+
+*iterating incoming unidirectional stream(s)*
+
+```rust
+async fn receive_unidirectional_stream(mut connection: NewConnection) -> anyhow::Result<()> {
+    while let Some(Ok(recv)) = connection.uni_streams.next().await {
+        // Because it is a unidirectional stream, we can only receive not sent back.
+        println!("{:?}", recv.read_to_end(50).await?);
+    }
+
+    Ok(())
+}
+```
+
+## Unreliable Messaging
+
+With unreliable messaging you can transfer data over UDP without reliability. 
+This could be useful if data-arrival isnt essential or when needing of high-throughput matters and reliab
```

Comment:
Unfinished sentence?

```suggestion
This could be useful if data arrival isn't essential or when needing of high throughput matters and reliab
```